### DISCUSSION TITLE
How do I get the request path with placeholders?

### DISCUSSION THREAD

**Discussion by jahands** [honojs/hono#1262] [Q&A] [answered]
Context: Tracing in Sentry is done per transaction - this is typically something like `/api/users/:user` (with the placeholder rather than an actual user ID.)

Given the following route:
```typescript
app.get('/api/users/:user', async (c) => {
    ...
})
```
Is it possible to get the string `/api/users/:user` from the request object?

**Comment by yusukebe** [comment]
Hi @jahands !

With the current implementation, the answer is "we can't do it". Because the result of the matching routing only has the "handler" and "named path param (user ID)", it does not have `/api/users/:user`. To make it possible, we would have to add some code to the routers. These are challenging aspects. So, we can't implement it right now, but we will leave this question open for future consideration. Thanks!

**Comment by scttcper** [comment]
Would love to see this implemented since its how you'd group requests in APM monitoring middleware like sentry. It's also pretty useful in just regular logging.

edit: just realized i've said the same thing jahands did

**Comment by scttcper** [comment, marked as answer]
It looks like this is available in [v3.11.0](https://github.com/honojs/hono/releases/tag/v3.11.0) via `c.req.routePath` thanks @usualoma

**Reply by jahands** [reply]
This works great! Thanks!
